components/camel-stitch/src/main/java/org/apache/camel/component/stitch/StitchConfiguration.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.stitch;
+
+import java.util.Collection;
+import java.util.HashSet;
+
+import org.apache.camel.RuntimeCamelException;
+import org.apache.camel.component.stitch.client.StitchClient;
+import org.apache.camel.component.stitch.client.StitchRegion;
+import org.apache.camel.component.stitch.client.models.StitchSchema;
+import org.apache.camel.spi.Metadata;
+import org.apache.camel.spi.UriParam;
+import org.apache.camel.spi.UriParams;
+import org.apache.camel.spi.UriPath;
+import reactor.netty.http.client.HttpClient;
+import reactor.netty.resources.ConnectionProvider;
+
+@UriParams
+public class StitchConfiguration implements Cloneable {
+
+    @UriPath
+    private String tableName;
+    @UriParam(label = "security", secret = true)
+    @Metadata(required = true)
+    private String token;
+    @UriParam(label = "producer", defaultValue = "europe")
+    private StitchRegion region = StitchRegion.EUROPE;
+    @UriParam(label = "producer")
+    @Metadata(autowired = true)
+    private StitchSchema stitchSchema;
+    @UriParam(label = "producer")
+    private Collection<String> keyNames = new HashSet<>();
```

Comment:
It would be good to also allow to set keyNames as a string value, where they keys are seperated with comma, then its easier to configure in endpoint uri. Some components offer that.